File: app/src/main/java/com/kt/kol/app/security/JwtUserPrincipal.java
package com.kt.kol.app.security;

import java.util.List;

public record JwtUserPrincipal(
    String userId,
    String username,
    List<String> roles
) {

    public boolean hasRole(String role) {
        return roles.contains(role);
    }

    public boolean hasAnyRole(String... roles) {
        for (String role : roles) {
            if (this.roles.contains(role)) {
                return true;
            }
        }
        return false;
    }
}

File: app/src/main/kotlin/com/kt/kol/app/security/JwtUserPrincipal.kt
package com.kt.kol.app.security

data class JwtUserPrincipal(
    val userId: String,
    val username: String,
    val roles: List<String>
) {
    fun hasRole(role: String): Boolean {
        return roles.contains(role)
    }

    fun hasAnyRole(vararg roles: String): Boolean {
        return roles.any { this.roles.contains(it) }
    }
}